setup/steps/set_up_monitoring_user.go

```diff
@@ -0,0 +1,72 @@
+package steps
+
+import (
+	"errors"
+	"fmt"
+
+	survey "github.com/AlecAivazis/survey/v2"
+	"github.com/lib/pq"
+	"github.com/pganalyze/collector/setup/query"
+	s "github.com/pganalyze/collector/setup/state"
+)
+
+// ensure user has correct permissions
+var SetUpMonitoringUser = &s.Step{
+	Description: "Set up monitoring user",
+	Check: func(state *s.SetupState) (bool, error) {
+		pgaUserKey, err := state.CurrentSection.GetKey("db_username")
+		if err != nil {
+			return false, err
+		}
+		pgaUser := pgaUserKey.String()
+
+		// TODO: deal with postgres <10
+		row, err := state.QueryRunner.QueryRow(
+			fmt.Sprintf(
+				"SELECT usesuper OR pg_has_role(usename, 'pg_monitor', 'member') FROM pg_user WHERE usename = %s",
```

Comment:
As discussed, we might want to have `pg_has_role(usename, 'pg_monitor', 'usage')` here instead